stories/query-rule-context.stories.ts

```diff
@@ -0,0 +1,63 @@
+import { storiesOf } from '@storybook/html';
+import { withHits } from '../.storybook/decorators';
+import moviesPlayground from '../.storybook/playgrounds/movies';
+
+type CustomDataItem = {
+  title: string;
+  banner: string;
+  link: string;
+};
+
+const searchOptions = {
+  appId: 'latency',
+  apiKey: 'af044fb0788d6bb15f807e4420592bc5',
+  indexName: 'instant_search_movies',
+  playground: moviesPlayground,
+};
+
+storiesOf('QueryRuleContext', module).add(
+  'default',
+  withHits(({ search, container, instantsearch }) => {
+    const widgetContainer = document.createElement('div');
+    const description = document.createElement('ul');
+    description.innerHTML = `
+        <li>Select the "Drama" category and The Shawshank Redemption appears</li>
+        <li>Select the "Thriller" category and Pulp Fiction appears</li>
+        <li>Type <q>music</q> and a banner will appear.</li>
+      `;
+
+    container.appendChild(description);
+    container.appendChild(widgetContainer);
+
+    search.addWidget(
+      instantsearch.widgets.queryRuleContext({
+        container,
+        trackedFilters: {
+          genre: () => ['Thriller', 'Drama'],
+        },
+      })
+    );
+
+    search.addWidget(
+      instantsearch.widgets.queryRuleCustomData({
+        container: widgetContainer,
+        transformItems: (items: CustomDataItem[]) => items[0],
```

Comment:
this story proves it was good that we put this API here, otherwise we'd have two banners visible. Deciding which one to choose is still totally easy to do and in user land 🙌